packages/gatsby-plugin-mdx/utils/gray-matter/index.js

```diff
@@ -0,0 +1,11 @@
+const grayMatter = require(`gray-matter`)
+
+const stripExcerpt = (excerpt, _options) => excerpt.trim().replace(/\n+/g, ` `)
```

Comment:
question: is stripping new lines per default right here or should it return 1:1 the excerpt and its up to the user to change it as needed?

how is it done in [packages/gatsby-transformer-remark](https://github.com/gatsbyjs/gatsby/tree/master/packages/gatsby-transformer-remark) since there is also an excerpt option there ?

Reply:
The `gatsby-transformer-remark` plugin [apparently uses `gray-matter`](https://using-remark.gatsbyjs.org/excerpts/) under the hood to retrieve the excerpt.

But if I'm understanding [the actual implementation](https://github.com/gatsbyjs/gatsby/blob/69ecf6a534b2be20e3b1ffe0b694ea677b68bb66/packages/gatsby-transformer-remark/src/extend-node-type.js#L436-L439) correctly, the `gray-matter` excerpt is only used in one particular case: when the `format:MARKDOWN` option is specified in the GraphQL query _and_ the `excerpt_separator` option is configured; in all other cases, the excerpt parsing is hand-rolled.

A quick test on my own site shows the behavior of the different excerpt formatting modes when an `excerpt_separator` is specified and the excerpt is rendered in a `pre` tag (to preserve whitespace and prevent text-wrapping).

| `PLAIN` | `MARKDOWN` | `HTML` |
:--------:|:----------:|:-------:
<img width="1680" alt="Screen Shot 2020-01-05 at 21 21 06" src="https://user-images.githubusercontent.com/2475919/71785601-ca93b600-3001-11ea-8666-ddb5b39f6bbe.png"> | <img width="1680" alt="Screen Shot 2020-01-05 at 21 21 16" src="https://user-images.githubusercontent.com/2475919/71785602-ca93b600-3001-11ea-8b15-4d096a9e7d3c.png"> | <img width="1680" alt="Screen Shot 2020-01-05 at 21 21 24" src="https://user-images.githubusercontent.com/2475919/71785603-ca93b600-3001-11ea-9c18-2dd0a76162d1.png">

The results are interesting. It seems `gatsby-transformer-remark` plugin is trying to preserve newlines (though in the case of `PLAIN` I think there is a slight bug: entirely blank lines cause the preceding and following line to be joined).

So if we are trying to mirror the behavior of the `Remark` plugin, I think we should preserve the output of `gray-matter` exactly, and let the user decide if they want to manipulate it further.

Reply:
Updated.